### PR TITLE
Replace win_fopen.c with wrappers used in Chocolate Doom

### PR DESCRIPTION
**Pull request by rfomin**
Fixes `autoload` and environmental variables (`DOOMWADDIR` etc) with non-Latin paths.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,4 +14,3 @@ cmake_install.cmake
 /prboom2/prboom-plus
 /prboom2/prboom-plus-game-server
 /prboom2/prboom-plus.wad
-/prboom2/build*
```
Comment (fabiangreffrath) [inline .gitignore:17]: Why this?

**Review by fabiangreffrath — APPROVED** [review]
Thank you!

File: .gitignore (hunk 1 above)
Comment (rfomin) [reply .gitignore:17]: I added this some time ago by mistake when I was implementing `win_fopen.c`.

**Comment by fabiangreffrath** [comment]
MacOS fails.

File: .gitignore (hunk 1 above)
Comment (fabiangreffrath) [reply .gitignore:17]: Ah, okay.

**Comment by rfomin** [comment]
> MacOS fails.

Fixed in https://github.com/coelckers/prboom-plus/pull/554/commits/c7d745103ed7ab2d4e71350eef3a94facd9f884a. Also I successfully built it in MSYS2.

**Comment by fabiangreffrath** [comment]
Great, thanks again.

**Pull request merged**